### ISSUE TITLE
Memory Leak in 1.3.2

### ISSUE THREAD

**Issue by birdonwheels5** [zcoinofficial/ccminer#74]
I've been running this on my Windows 7 GTX 1080 TI system for a few days now and after running for a few hours I notice the memory usage steadily climbs, and eventually the miner crashes. I noticed this for two nights in a row and it's frustrating to wake up to missed blocks.

I just put together a quick python script that will mitigate the issue for now (starting/stopping the miner every two hours), but I'm surprised others haven't run into this.

I'm mining FIRO with the MTP algorithm and the standard ccminer.exe binary for release 1.3.2. Let me know if you need any additional information and I can try to help you get it.

**Comment by djm34** [comment]
can you the previous version ? 
Memory leak have been solved long ago, so may-be it is something which was introduced later on

**Comment by djm34** [comment]
was re-reading your message, are you running windows 7 or 7 1080ti ?

**Comment by birdonwheels5** [comment]
Sure, I'll give the previous message a shot, and it's Windows 7 with only one 1080 TI, sorry for the confusion.

**Comment by birdonwheels5** [comment]
So I tried the previous version and I was getting like 10x slower speeds from that one, with the same launch settings. So unfortunately I'm going to have to go back to the newer one and deal with the leak.

I did manage to check task manager when it was ramping up memory usage and I saw ccminer sitting at 16+ GB of memory usage. It doesn't really ramp up gradually, it's after a while it decides to start accumulating until it crashes. It would explain the 90+ GB page file I was seeing a few days ago before I disabled the page file (OS is on an SSD).

**Comment by djm34** [comment]
have you tried the version called ccminer_cuda11.exe ?
https://github.com/firoorg/ccminer/releases/download/1.3.2/ccminer_cuda11.exe
this is the latest (only difference is that it support 30xx card...)

What you say doesn't make too much sense, or may-be it is because you are using windows7. 
I ran the program for about 45min, I noticed absolutely no change in page_file usage which remained all the time at 5111544ko (around 5Gb) and physical ram at 290340ko (around 290Mb). 

So basically there is no memory leak (and many hours of work went into memory leak testing in the last years...), and definitely nothing in the size of what you are reporting... so either you are using the wrong version (that's why I ask you to check that cuda11 version) or it is a problem with windows 7 as this version was build around windows 10.

**Comment by birdonwheels5** [comment]
Okay, I just tried it again and it's giving the same as the newer version. I must have accidentally had a mining process running in the background last I tried. I'll leave it on and see how it goes, and let you know.

**Comment by birdonwheels5** [comment]
Also to your point I think I've been using the version for cuda 10. I downloaded and tried the one you linked to. I'll let that one run and let you know if I run into a memory leak.

**Comment by birdonwheels5** [comment]
Okay, so it seems to be stable the past few days - granted, I have been using my script and killing/restarting every 15 minutes. I have not had any messages saying it has run out of memory in that time.

We can close this for now. If I have time I can try to run it without the management script to see if it crashes while I'm at work, but I will have to keep a close eye on it in case it does go down. The past week has been very busy both in my personal and professional life, so I haven't had time to keep a close eye on it.

**Comment by djm34** [comment]
15min kill ... I ran for over 30min, nothing at all had changed in terms of memory usage...

**Comment by birdonwheels5** [comment]
There definitely is a problem because I've been encountering issues the past week, and last night it ran out of memory again. When I started last night I was at 3GB memory used. I woke up this morning and there was 17.7GB used, and all I had open was one web browser window and the miner. The miner also crashed and stopped mining at 1:30 AM. Maybe today I will try installing Linux and getting it set up there, but in my experience getting display drivers working in Linux is painful. I had my script killing ccminer less aggressively at every 30 min instead of 15. I tried with every hour the other day, but same memory results.

I'm getting the error "The instruction at 0xf70d13ba referenced memory at 0x00000000. The memory could not be written. Click on OK to terminate the program." This memory has been mem tested for a pass with no errors because I was going to sell it, but I had some extra so I threw it in my mining computer.

**Comment by djm34** [comment]
upgrade to windows 10. There is no such problem with windows 10
